src/config.c

```diff
@@ -1807,3 +1857,236 @@ int cgroup_config_create_template_group(struct cgroup *cgroup, char *template_na
 	cgroup_free(&aux_cgroup);
 	return ret;
 }
+
+#ifdef WITH_SYSTEMD
+int cgroup_add_systemd_opts(const char * const config, const char * const value)
+{
+	struct cgroup_systemd_opts *curr = cgroup_systemd_opts_tail;
+	int len;
+
+	if (strcmp(config, "slice") == 0) {
+		snprintf(curr->slice_name, FILENAME_MAX, "%s", value);
+
+		len = strlen(curr->slice_name) - 6;
+		if (strcmp(curr->slice_name + len, ".slice"))
+			goto err;
+
+	} else if (strcmp(config, "scope") == 0) {
+		snprintf(curr->scope_name, FILENAME_MAX, "%s", value);
+
+		len = strlen(curr->scope_name) - 6;
+		if (strcmp(curr->scope_name + len, ".scope"))
+			goto err;
+
+	} else if (strcmp(config, "setdefault") == 0 && strcasecmp(value, "yes") == 0) {
+		curr->setdefault = 1;
+	} else if (strcmp(config, "pid") == 0) {
+		/*
+		 * If the atoi() fails, the pid value is zero and also
+		 * avoid allowing init task (systemd).
+		 */
+		curr->pid = atoi(value);
+		if (curr->pid <= 1)
+			goto err;
+	} else
+		goto err;
+
+	return 1;
+err:
+	cgroup_err("Invalid systemd configuration %s value %s\n", config, value);
+	cgroup_cleanup_systemd_opts();
+	return 0;
+}
+
+int cgroup_alloc_systemd_opts(const char * const config, const char * const value)
+{
+	struct cgroup_systemd_opts *new_cgrp_systemd_opts;
+
+	/*
+	 * check for the allowed systemd configurations. We don't
+	 * check for values, they will be checked by systemd anyway.
+	 */
+	if (strcmp(config, "slice") != 0 &&
+	    strcmp(config, "scope") != 0 &&
+	    strcmp(config, "setdefault") != 0 &&
```

Comment:
My apologies up front.... :)

Now that I think about it more, would naming this parameter `default` (rather than `setdefault`) make more sense?  We didn't name the other parameters `setslice`, `setscope`, etc. but they're obviously setting something.

Reply:
I completely agree, in fact, I had thought about renaming it to `default` will rename it.

Reply:
actually, `default` can't be used as its a keyword, does `delegate` sound better?

Reply:
Argh... thanks for digging.  Let's stick with `setdefault` then.  `delegate` is likely an option we will need to add later, but it should apply to the `opts->delegate` field within the systemd options structure.

Reply:
Ah, Make sense, will leave it unchanged.